test/e2e/provider_aws_test.go

```diff
@@ -0,0 +1,189 @@
+	"github.com/Mirantis/hmc/test/e2e/managedcluster/clusteridentity"
+	"github.com/Mirantis/hmc/test/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("AWS Templates", Label("provider:cloud", "provider:aws"), Ordered, func() {
+	var (
+		kc                   *kubeclient.KubeClient
+		standaloneClient     *kubeclient.KubeClient
+		standaloneDeleteFunc func() error
+		hostedDeleteFunc     func() error
+		kubecfgDeleteFunc    func() error
+		clusterName          string
+	)
+
+	BeforeAll(func() {
+		By("providing cluster identity")
+		kc = kubeclient.NewFromLocal(internalutils.DefaultSystemNamespace)
+		ci := clusteridentity.New(kc, managedcluster.ProviderAWS)
+		Expect(os.Setenv(managedcluster.EnvVarAWSClusterIdentity, ci.IdentityName)).Should(Succeed())
+	})
+
+	AfterAll(func() {
+		// If we failed collect logs from each of the affiliated controllers
+		// as well as the output of clusterctl to store as artifacts.
+		if CurrentSpecReport().Failed() && !noCleanup() {
+			if standaloneClient != nil {
+				By("collecting failure logs from hosted controllers")
+				collectLogArtifacts(standaloneClient, clusterName, managedcluster.ProviderAWS, managedcluster.ProviderCAPI)
+			}
+		}
+
+		By("deleting resources")
+		for _, deleteFunc := range []func() error{
+			kubecfgDeleteFunc,
+			hostedDeleteFunc,
+			standaloneDeleteFunc,
+		} {
+			if deleteFunc != nil {
+				err := deleteFunc()
+				Expect(err).NotTo(HaveOccurred())
+			}
+		}
+	})
+
+	It("should work with an AWS provider", func() {
+		// Deploy a standalone cluster and verify it is running/ready.
+		// Deploy standalone with an xlarge instance since it will also be
+		// hosting the hosted cluster.
+		GinkgoT().Setenv(managedcluster.EnvVarAWSInstanceType, "t3.xlarge")
+		GinkgoT().Setenv(managedcluster.EnvVarInstallBeachHeadServices, "false")
+
+		templateBy(managedcluster.TemplateAWSStandaloneCP, "creating a ManagedCluster")
+		sd := managedcluster.GetUnstructured(managedcluster.TemplateAWSStandaloneCP)
+		clusterName = sd.GetName()
+
+		standaloneDeleteFunc = kc.CreateManagedCluster(context.Background(), sd)
+
+		templateBy(managedcluster.TemplateAWSStandaloneCP, "waiting for infrastructure to deploy successfully")
+		deploymentValidator := managedcluster.NewProviderValidator(
+			managedcluster.TemplateAWSStandaloneCP,
+			clusterName,
+			managedcluster.ValidationActionDeploy,
+		)
+
+		Eventually(func() error {
+			return deploymentValidator.Validate(context.Background(), kc)
+		}).WithTimeout(30 * time.Minute).WithPolling(10 * time.Second).Should(Succeed())
+
+		templateBy(managedcluster.TemplateAWSHostedCP, "installing controller and templates on standalone cluster")
+
+		// Download the KUBECONFIG for the standalone cluster and load it
+		// so we can call Make targets against this cluster.
+		// TODO: Ideally we shouldn't use Make here and should just convert
+		// these Make targets into Go code, but this will require a
+		// helmclient.
```

Comment:
I'm suggesting to create actual tasks and reference it here, so it will be _actually_ done sometime in the future

Reply:
Ya that's fair, I'll create them.